pylot/map/hd_map.py

```diff
@@ -334,24 +338,63 @@ def get_lane(self, location: Location, waypoint_precision: float = 0.05):
             self._lateral_shift(w.transform, w.lane_width * 0.5)
             for w in lane_waypoints
         ]
-        return Lane(left_markings, right_markings)
+        return Lane(lane_id, left_markings, right_markings)
 
-    def get_left_lane(self, location: Location):
-        waypoint = self._get_waypoint(location, project_to_road=False)
+    def get_left_lane(self, location):
+        waypoint = self._get_waypoint(location,
+                                      project_to_road=False,
+                                      lane_type=carla.LaneType.Any)
         if waypoint:
             left_lane_waypoint = waypoint.get_left_lane()
             if left_lane_waypoint:
-                return Transform.from_carla_transform(waypoint.transform)
+                return Transform.from_carla_transform(
+                    left_lane_waypoint.transform)
         return None
 
-    def get_right_lane(self, location: Location):
-        waypoint = self._get_waypoint(location, project_to_road=False)
+    def get_right_lane(self, location):
+        waypoint = self._get_waypoint(location,
+                                      project_to_road=False,
+                                      lane_type=carla.LaneType.Any)
         if waypoint:
             right_lane_waypoint = waypoint.get_right_lane()
             if right_lane_waypoint:
-                return Transform.from_carla_transform(waypoint.transform)
+                return Transform.from_carla_transform(
+                    right_lane_waypoint.transform)
         return None
 
+    def get_all_lanes(self, location):
+        lanes = [self.get_lane(location)]
+
+        waypoint = self._get_waypoint(location,
+                                      project_to_road=False,
+                                      lane_type=carla.LaneType.Any)
+        if waypoint:
+            wp_left = waypoint.get_left_lane()
+            w_rotation = waypoint.transform.rotation
+            while wp_left and wp_left.lane_type == carla.LaneType.Driving:
+                camera_transform = Transform.from_carla_transform(
+                    wp_left.transform)
+                lanes.append(
+                    self.get_lane(camera_transform.location,
+                                  lane_id=wp_left.lane_id))
```

Comment:
You don't need to build the Pylot transform out of the CARLA transform because you're only using the location. You can instead do `left_location = Location.from_carla_location(wp_left.transform.location)`